nifi-extension-bundles/nifi-poi-bundle/nifi-poi-processors/src/main/java/org/apache/nifi/processors/excel/SplitExcel.java

```diff
@@ -0,0 +1,239 @@
+@WritesAttributes({
+        @WritesAttribute(attribute = "fragment.identifier", description = "All split Excel FlowFiles produced from the same parent Excel FlowFile will have the same randomly generated UUID added" +
+                " for this attribute"),
+        @WritesAttribute(attribute = "fragment.index", description = "A one-up number that indicates the ordering of the split Excel FlowFiles that were created from a single parent Excel FlowFile"),
+        @WritesAttribute(attribute = "fragment.count", description = "The number of split Excel FlowFiles generated from the parent Excel FlowFile"),
+        @WritesAttribute(attribute = "segment.original.filename", description = "The filename of the parent Excel FlowFile"),
+        @WritesAttribute(attribute = SplitExcel.SHEET_NAME, description = "The name of the Excel sheet from the original spreadsheet."),
+        @WritesAttribute(attribute = SplitExcel.TOTAL_ROWS, description = "The number of rows in the Excel sheet from the original spreadsheet.")})
+public class SplitExcel extends AbstractProcessor {
+    public static final String FRAGMENT_ID = FragmentAttributes.FRAGMENT_ID.key();
+    public static final String FRAGMENT_INDEX = FragmentAttributes.FRAGMENT_INDEX.key();
+    public static final String FRAGMENT_COUNT = FragmentAttributes.FRAGMENT_COUNT.key();
+    public static final String SEGMENT_ORIGINAL_FILENAME = FragmentAttributes.SEGMENT_ORIGINAL_FILENAME.key();
+    public static final String SHEET_NAME = "sheetname";
+    public static final String TOTAL_ROWS = "total.rows";
+
+    public static final PropertyDescriptor PROTECTION_TYPE = new PropertyDescriptor
+            .Builder().name("Protection Type")
+            .displayName("Protection Type")
+            .description("Specifies whether an Excel spreadsheet is protected by a password or not.")
+            .required(true)
+            .allowableValues(ProtectionType.class)
+            .defaultValue(ProtectionType.UNPROTECTED)
+            .build();
+
+    public static final PropertyDescriptor PASSWORD = new PropertyDescriptor
+            .Builder().name("Password")
+            .displayName("Password")
+            .description("The password for a password protected Excel spreadsheet")
+            .required(true)
+            .sensitive(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .dependsOn(PROTECTION_TYPE, ProtectionType.PASSWORD)
+            .build();
+
+    public static final PropertyDescriptor AVOID_TEMP_FILES = new PropertyDescriptor
+            .Builder().name("Avoid Temp Files")
+            .displayName("Avoid Temp Files")
+            .description("Enables a mode where the code tries to avoid creating temp files. By default, temp files are used to avoid holding onto too much data in memory.")
+            .required(true)
+            .defaultValue(Boolean.FALSE.toString())
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
```

Comment:
I recommend removing this property since it should always be enabled for general processing.

Reply:
I need it for `testPasswordProtected` in `TestSplitExcel`. Should I remove that test?

Reply:
If that is the only way to exercise that test, then yes, I would remove the property and the test method. I know this was an issue for the previous implementation of the password-protection in ExcelReader, but exposing a Processor property just for testing is less than ideal, so removing it sounds like the best option under the circumstances.